build/lib/src/asset/id.dart

```diff
@@ -110,9 +110,9 @@ class AssetId implements Comparable<AssetId> {
   /// Note that this is intended for communicating ids across isolates and not
   /// for persistent storage of asset identifiers. There is no guarantee of
   /// backwards compatibility in serialization form across versions.
-  AssetId.deserialize(data)
-      : package = data[0],
-        path = data[1];
+  AssetId.deserialize(List data)
```

Comment:
Yeah today we generally have to cast to `List` but we'd have to cast to `List<String>` if I change the argument - and that cast will fail with strong mode runtime semantics - so we'd have to do `new List<String>.from(values)` when we know that `values` only contains `String` elements... that is really clunky so I prefer the cast at the point where we pull the elements out (that's similar to what the from ctor does - shifts collection level generic to element level)